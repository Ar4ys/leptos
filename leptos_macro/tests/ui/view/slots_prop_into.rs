use leptos::*;

#[slot]
struct Then {
    children: ChildrenFn,
}

#[slot]
struct ElseIf {
    #[prop(into)]
    cond: MaybeSignal<bool>,
    children: ChildrenFn,
}

#[component]
fn SlotIf(
    #[prop(into)] cond: MaybeSignal<bool>,
    then: Then,
    #[prop(optional)] else_if: Vec<ElseIf>,
) -> impl IntoView {
    _ = cond;
    _ = then;
    _ = else_if;
}

fn main() {
    let (count, set_count) = create_signal(0);
    let is_even = MaybeSignal::derive(move || count.get() % 2 == 0);
    let is_div5 = move || count.get() % 5 == 0;

    view! {
        <SlotIf cond=is_even>
            <Then slot>"even"</Then>
            <ElseIf slot cond=is_div5>"divisible by 5"</ElseIf>
        </SlotIf>
    };
}
