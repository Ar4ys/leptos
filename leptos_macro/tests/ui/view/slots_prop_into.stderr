error[E0277]: the trait bound `leptos::MaybeSignal<bool>: From<{closure@$DIR/tests/ui/view/slots_prop_into.rs:29:19: 29:26}>` is not satisfied
  --> tests/ui/view/slots_prop_into.rs:34:31
   |
34 |             <ElseIf slot cond=is_div5>"divisible by 5"</ElseIf>
   |                          ---- ^^^^^^^ the trait `From<{closure@$DIR/tests/ui/view/slots_prop_into.rs:29:19: 29:26}>` is not implemented for `leptos::MaybeSignal<bool>`
   |                          |
   |                          required by a bound introduced by this call
   |
   = help: the following other types implement trait `From<T>`:
             <leptos::MaybeSignal<std::string::String> as From<&str>>
             <leptos::MaybeSignal<T> as From<ReadSignal<T>>>
             <leptos::MaybeSignal<T> as From<RwSignal<T>>>
             <leptos::MaybeSignal<T> as From<Memo<T>>>
             <leptos::MaybeSignal<T> as From<Signal<T>>>
             <leptos::MaybeSignal<T> as From<T>>
   = note: required for `{closure@$DIR/tests/ui/view/slots_prop_into.rs:29:19: 29:26}` to implement `Into<leptos::MaybeSignal<bool>>`
note: required by a bound in `ElseIfBuilder::<((), __children)>::cond`
  --> tests/ui/view/slots_prop_into.rs:8:1
   |
8  | #[slot]
   | ^^^^^^^ required by this bound in `ElseIfBuilder::<((), __children)>::cond`
...
11 |     cond: MaybeSignal<bool>,
   |     ---- required by a bound in this associated function
   = note: this error originates in the derive macro `::leptos::typed_builder_macro::TypedBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
